internal/commands/vaultsecrets/secrets/create.go

```diff
@@ -98,35 +120,177 @@ type CreateOpts struct {
 	SecretName           string
 	SecretValuePlaintext string
 	SecretFilePath       string
+	Type                 SecretType
 	PreviewClient        preview_secret_service.ClientService
 	Client               secret_service.ClientService
 }
 
+type SecretConfig struct {
+	Version         string
+	Type            integrations.IntegrationType
+	IntegrationName string `yaml:"rotation_integration_name"`
+	Details         map[string]any
+}
+
+type MongoDBRole struct {
+	RoleName       string `mapstructure:"role_name"`
+	DatabaseName   string `mapstructure:"database_name"`
+	CollectionName string `mapstructure:"collection_name"`
+}
+
+type MongoDBScope struct {
+	Name string `mapstructure:"type"`
+	Type string `mapstructure:"name"`
+}
+
+var (
+	TwilioKeys = []string{"rotation_policy_name"}
+	MongoKeys  = []string{"rotation_policy_name", "mongodb_group_id"}
```

Comment:
rotation_policy_name should be required for all rotating secrets (just like `rotation_integration_name`

For MongoDB, roles is also required

Reply:
`rotation_integration_name` is added here for these two separately because the `SecretConfig` is shared w/ dynamic secrets as well, which [don't have `rotation_integration_name`](https://github.com/hashicorp/hcp/pull/143/files#diff-7f92b17a95650abeca3ec432e8bcf6859f19709d1bdb6b3687fb5d704940a4ecR146-R150). I could see an argument for making a separating the two though.

Reply:
Gotcha, it's not a big deal then. If any required parameter is missing, the request will be rejected by server side validation